doc/user/interop.md

```diff
@@ -75,6 +79,12 @@ Calls `to_native` if the object responds to it, otherwise throws
 
 Returns true only for the `nil` object.
 
+### `HAS_KEYS`
+
+Returns true for all objects except those primitives that are always frozen,
```

Comment:
I think it would be good to document the intent of this message here. It's good that you indicate how it behaves, but it's not immediately clear why anything else would have "keys".

Reply:
Maybe it's good enough to just note that this is the complement to the `KEYS` message. But I think leading with this section before the `KEYS` section doesn't really link the two that well.

Reply:
Interop messages are already documented elsewhere. I don't think it's a good idea to just repeat that documentation in this document which is about how Ruby responds to interop and how to send interop messages using Ruby - maybe I should just link to the original documentation as well.

Reply:
:+1: for a link to the JavaDoc at the top.

Reply:
Done.